src/ir/ty.rs

```diff
@@ -143,17 +127,15 @@ impl Type {
             TypeKind::BlockPointer |
             TypeKind::Int(..) |
             TypeKind::Float(..) |
-            TypeKind::Named(..) => true,
+            TypeKind::Named => true,
             _ => false,
         }
     }
 
     /// Creates a new named type, with name `name`.
     pub fn named(name: String) -> Self {
         assert!(!name.is_empty());
```

Comment:
Maybe it would be good to assert about identifiers like we now do in codegen here as well?